tests/ClerkGuardTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace RonasIT\Clerk\Tests;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Config;
+use RonasIT\Clerk\Auth\ClerkGuard;
+use RonasIT\Clerk\Exceptions\EmptyConfigException;
+use RonasIT\Clerk\Tests\Support\TokenMockTrait;
+
+class ClerkGuardTest extends TestCase
+{
+    use TokenMockTrait;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        Config::set('clerk', [
+            'allowed_issuer' => 'issuer',
+            'secret_key' => self::SECRET_KEY_PASS,
+            'signer_key_path' => self::SIGNER_KEY_PATH,
+        ]);
+    }
+
+    public function testAuthUser(): void
+    {
+        $clerkToken = $this
+            ->createJWTToken('user_id')
+            ->toString();
+
+        $request = new Request();
+        $request->headers->set('Authorization', "Bearer {$clerkToken}");
```

Comment:
let's move it to the helper


```suggestion
        $request = $this->generateRequest(['Authorization' => "Bearer {$clerkToken}"]);
```